### ISSUE TITLE
[BUG] Long loop error with warning notification

### ISSUE THREAD

**Issue by apupier** [fabric8-analytics/fabric8-analytics-vscode-extension#700]
**Describe the bug**

I have a long loop with a notification warning popping up almost every seconds with text `Cannot convert undefined or null to object`

**To Reproduce**

Do not know exactly, it happened several times today with my this project https://github.com/camel-tooling/camel-dap-client-vscode/
1. Open the folder on https://github.com/camel-tooling/camel-dap-client-vscode/
2. and i do not know

**Expected behavior**

**Screenshots**

**Logs**
```
Error: Cannot find module '/tmp/exhort_eGy7My/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_eGy7My
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_11_18_184Z-debug-0.log
[Warn  - 10:11:44 AM] Component Analysis Error: Error: Command failed: npm i --package-lock-only --prefix /tmp/exhort_eGy7My
node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/tmp/exhort_eGy7My/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_eGy7My
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_11_18_184Z-debug-0.log

node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/tmp/exhort_uOy2n8/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_uOy2n8
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_15_40_946Z-debug-0.log
[Warn  - 10:16:30 AM] Component Analysis Error: Error: Command failed: npm i --package-lock-only --prefix /tmp/exhort_uOy2n8
node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/tmp/exhort_uOy2n8/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_uOy2n8
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_15_40_946Z-debug-0.log

node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/tmp/exhort_PH7ZP1/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_PH7ZP1
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_16_32_013Z-debug-0.log
[Warn  - 10:16:44 AM] Component Analysis Error: Error: Command failed: npm i --package-lock-only --prefix /tmp/exhort_PH7ZP1
node:internal/modules/cjs/loader:1146
  throw err;
  ^

Error: Cannot find module '/tmp/exhort_PH7ZP1/scripts/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1143:15)
    at Module._load (node:internal/modules/cjs/loader:984:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.12.0
npm ERR! code 1
npm ERR! path /tmp/exhort_PH7ZP1
npm ERR! command failed
npm ERR! command sh -c node ./scripts/postinstall.js

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_16_32_013Z-debug-0.log

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: Code@1.82.2
npm ERR! Found: xterm@5.3.0-beta.61
npm ERR! node_modules/xterm
npm ERR!   xterm@"5.3.0-beta.61" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer xterm@"^5.0.0" from xterm-addon-canvas@0.5.0-beta.22
npm ERR! node_modules/xterm-addon-canvas
npm ERR!   xterm-addon-canvas@"0.5.0-beta.22" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/apupier/.npm/_logs/2024-04-11T08_16_46_692Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_16_46_692Z-debug-0.log
[Warn  - 10:17:14 AM] Component Analysis Error: Error: Command failed: npm i --package-lock-only --prefix /tmp/exhort_Yg8dz6
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: Code@1.82.2
npm ERR! Found: xterm@5.3.0-beta.61
npm ERR! node_modules/xterm
npm ERR!   xterm@"5.3.0-beta.61" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer xterm@"^5.0.0" from xterm-addon-canvas@0.5.0-beta.22
npm ERR! node_modules/xterm-addon-canvas
npm ERR!   xterm-addon-canvas@"0.5.0-beta.22" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/apupier/.npm/_logs/2024-04-11T08_16_46_692Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /home/apupier/.npm/_logs/2024-04-11T08_16_46_692Z-debug-0.log

[Warn  - 10:17:16 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:18 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:20 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:23 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:26 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:28 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:30 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:32 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:34 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:36 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:38 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:40 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:43 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:45 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:48 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:50 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:52 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:55 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:17:59 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:03 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:07 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:10 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:16 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:21 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:28 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:35 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:38 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:41 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:44 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:46 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:49 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:51 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:54 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:56 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:18:58 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:00 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:09 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:13 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:16 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:18 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:19 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:21 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:23 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:25 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:27 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:29 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:31 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:36 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:40 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:43 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:45 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:47 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:49 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:51 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:53 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:55 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:57 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:19:59 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:01 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:03 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:06 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:08 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:11 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:13 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:16 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:19 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:21 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:24 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:26 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:29 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:31 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:33 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:35 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:37 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:39 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:42 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:44 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:46 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:48 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:50 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:52 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:54 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:56 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:20:59 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:21:05 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:21:07 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:21:10 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:21:18 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object
[Warn  - 10:21:20 AM] Component Analysis Error: TypeError: Cannot convert undefined or null to object

```

**VSCode(please complete the following information):**
 - OS: Fedora 39
 - VSCode version 1.88.0
 - Dependency Analytics Version 0.9.4

**CHE7/Theia(please complete the following information):**
 - OS: [e.g. macOS]
 - Browser [e.g. chrome]
 - CHE/Theia version 
 - Dependency Analytics Version [e.g. 0.0.22]

**Additional context**

**Comment by IlonaShishov** [comment]
Hello @apupier,

Thank you for bringing up this issue.

As a temporary solution, I suggest refraining from using relative paths (./scripts/postinstall.js) and opting for absolute paths instead. This should help mitigate the problem for the time being.

We're currently working on a new version that will include support for relative paths, which will be available soon.

**Comment by IlonaShishov** [comment]
Understood, @apupier. 
Thank you for your patience. Relative path support will be available soon.

**Comment by IlonaShishov** [comment]
Hello,

This issue has been resolved in the latest release, version 0.9.5. We appreciate your help in improving the project. If you have any further issues or suggestions, please feel free to reach out.

Best regards.
